src/App.js

```diff
@@ -14,11 +16,124 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isHidden: true,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  buttonSatus = () => {
+    this.setState({
+      isHidden: false,
+    });
+  }
+
+  reverseMethod = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortByField = (field) => {
+    this.setState({
+      sortBy: field,
+    });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  }
+
+  render() {
+    const {
+      goods,
+      isHidden,
+      isReversed,
+      sortBy,
+    } = this.state;
+
+    const copyOfGoods = [...goods];
+
+    copyOfGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'byAlphabet':
+          return a.localeCompare(b);
+
+        case 'byLength':
+          return a.length - b.length;
+        default:
+          return '';
+      }
+    });
+
+    if (isReversed) {
+      copyOfGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+        {isHidden && (
+          <button
+            type="button"
+            onClick={this.buttonSatus}
+          >
+            Start
+          </button>
+        )}
+
+        {!isHidden && (
+          <>
+            <ul className="list">
+              {copyOfGoods.map(el => (
+                <li key={el}>
+                  {el}
+                </li>
+              ))}
+            </ul>
+
+            <div className="buttons">
+              <button
+                type="button"
+                className={classNames(isReversed ? 'active' : '')}
```

Comment:
```suggestion
                className={classNames({'active' : isReversed})}
```
Use classnames library properly